frontend/src/services/ApiServiceFactory.ts

```diff
@@ -0,0 +1,45 @@
+import httpInstance from '@/services/http/HttpCommon';
+import {
+    FAMApplicationsApi,
+    FAMRolesApi,
+    FAMUserRoleAssignmentApi,
+    FAMUsersApi
+} from 'fam-api';
+
+export enum ApiServiceName {
+    ApplicationsApi = "ApplicationsApi",
+    RolesApi = "RolesApi",
+    UsersApi = "UsersApi",
+    UserRoleAssignmentApi = "UserRoleAssignmentApi",
+}
+
+export class ApiServiceFactory {
+    private applicationsApi: FAMApplicationsApi
+    private rolesApi: FAMRolesApi
+    private usersApi: FAMUsersApi
+    private userRoleAssignmentApi: FAMUserRoleAssignmentApi
+
+   constructor() {
+       // Instanciation for generated 'fam-api' client.
+       this.applicationsApi = new FAMApplicationsApi(undefined, '', httpInstance) // Note, Axios is strange, second parameter needs empty string, not null.
+       this.rolesApi = new FAMRolesApi(undefined, '', httpInstance)
+       this.usersApi = new FAMUsersApi(undefined, '', httpInstance)
+       this.userRoleAssignmentApi = new FAMUserRoleAssignmentApi(undefined, '', httpInstance)
+   }
+
+   getApiService(serviceName: ApiServiceName): FAMApplicationsApi | FAMRolesApi | FAMUsersApi | FAMUserRoleAssignmentApi {
```

Comment:
We could just have a separate getter for each service. I'm not sure what the advantage is of a consolidated method?

Reply:
Perhaps in this case I am used to Factory Pattern as it feels like it is just a Factory that produces an instance with different flavors (like Shape factory: produce Circle, Square...). There is no advantage or disadvantage difference than using getter as I can tell, but maybe needing a Type cast is an disadvantage, however class object is much cleaner since almost no logic at all. Do you like me to switch to getter, I don't mind, if you prefer that way?

Reply:
It is still factory pattern either way, just making it simpler for the caller to have the getters.
E.g. factory.getApplicationApi() vs factory.getService(ServiceName.ApplicationApi) as FamApplicationApi

Reply:
So I'd prefer the getters.

Reply:
No problem. I changed that so caller is easy to use this.